src/libraries/System.Net.Security/src/System/Net/Security/NetSecurityTelemetry.cs

```diff
@@ -0,0 +1,307 @@
+        public static readonly NetSecurityTelemetry Log = new NetSecurityTelemetry();
+
+        private IncrementingPollingCounter? _tlsHandshakeRateCounter;
+        private PollingCounter? _totalTlsHandshakesCounter;
+        private PollingCounter? _currentTlsHandshakesCounter;
+        private PollingCounter? _failedTlsHandshakesCounter;
+        private PollingCounter? _sessionsOpenCounter;
+        private PollingCounter? _sessionsOpenTls10Counter;
+        private PollingCounter? _sessionsOpenTls11Counter;
+        private PollingCounter? _sessionsOpenTls12Counter;
+        private PollingCounter? _sessionsOpenTls13Counter;
+        private EventCounter? _handshakeDurationCounter;
+        private EventCounter? _handshakeDurationTls10Counter;
+        private EventCounter? _handshakeDurationTls11Counter;
+        private EventCounter? _handshakeDurationTls12Counter;
+        private EventCounter? _handshakeDurationTls13Counter;
+
+        private long _finishedTlsHandshakes; // Successfully and failed
+        private long _startedTlsHandshakes;
+        private long _failedTlsHandshakes;
+        private long _sessionsOpen;
+        private long _sessionsOpenTls10;
+        private long _sessionsOpenTls11;
+        private long _sessionsOpenTls12;
+        private long _sessionsOpenTls13;
+
+        protected override void OnEventCommand(EventCommandEventArgs command)
+        {
+            if (command.Command == EventCommand.Enable)
+            {
+                _tlsHandshakeRateCounter ??= new IncrementingPollingCounter("tls-handshake-rate", this, () => Interlocked.Read(ref _finishedTlsHandshakes))
+                {
+                    DisplayName = "TLS handshakes completed",
+                    DisplayRateTimeScale = TimeSpan.FromSeconds(1)
+                };
+
+                _totalTlsHandshakesCounter ??= new PollingCounter("total-tls-handshakes", this, () => Interlocked.Read(ref _finishedTlsHandshakes))
+                {
+                    DisplayName = "Total TLS handshakes completed"
+                };
+
+                _currentTlsHandshakesCounter ??= new PollingCounter("current-tls-handshakes", this, () => -Interlocked.Read(ref _finishedTlsHandshakes) + Interlocked.Read(ref _startedTlsHandshakes))
+                {
+                    DisplayName = "Current TLS handshakes"
+                };
+
+                _failedTlsHandshakesCounter ??= new PollingCounter("failed-tls-handshakes", this, () => Interlocked.Read(ref _failedTlsHandshakes))
+                {
+                    DisplayName = "Total TLS handshakes failed"
+                };
+
+                _sessionsOpenCounter ??= new PollingCounter("all-sessions-open", this, () => Interlocked.Read(ref _sessionsOpen))
+                {
+                    DisplayName = "All Sessions Active"
+                };
+
+                _sessionsOpenTls10Counter ??= new PollingCounter("tls10-sessions-open", this, () => Interlocked.Read(ref _sessionsOpenTls10))
```

Comment:
Is there a use-case in mind for counters that distinguish each TLS protocol version separately? I was a little surprised to see these.

Reply:
Tomas outlined a possible scenario.
https://github.com/dotnet/runtime/pull/40108#discussion_r468308155

For handshake durations, I could see it being used to see if for some reason a certain tls version was taking much longer than others to set establish.

I don't know what exact scenarios David had in mind when listing these counters. To me they seem like nice-to-have, but not critical like the Start/Stop events here.

Reply:
> I don't know what exact scenarios David had in mind when listing these counters

David = @davidsh ? Perhaps we can lure him back to weigh in? : )

My speculation is that the value might be low. One of the things I hope we improve in .NET 6 is better support for multi-dimensional counters so we may get a better way to build this type of functionality if we wait a bit. However if this moves ahead as-is I don't have any significant concern. Thanks @MihaZupan!

Reply:
> One of the things I hope we improve in .NET 6 is better support for multi-dimensional counters so we may get a better way to build this type of functionality if we wait a bit

That sounds good, it would make this cleaner.

I believe there is value in having these in for YARP before 6.0.
I would prefer to get this change in master for 5.0 and if we decide in the next few days that it's okay to wait till 6.0 we can still remove them next week.